validator/src/main.rs

```diff
@@ -2666,91 +2768,88 @@ pub fn main() {
-        if addr.contains("https") {
-            backend_endpoint = backend_endpoint
-                .tls_config(tonic::transport::ClientTlsConfig::new())
-                .expect("failed to set tls_config");
-        }
-
-        BlockEngineConfig {
-            auth_service_endpoint,
-            backend_endpoint,
-            trust_packets: matches.is_present("trust_block_engine_packets"),
-        }
-    });
-
-    let is_relayer_enabled = matches.is_present("relayer_url")
-        || matches.is_present("relayer_auth_service_address")
-        || matches.is_present("relayer_address")
-        || matches.is_present("trust_relayer_packets")
-        || matches.is_present("relayer_expected_heartbeat_interval_ms")
-        || matches.is_present("relayer_max_failed_heartbeats");
-    let maybe_relayer_config = is_relayer_enabled.then(|| {
-        let addr: String = value_of(&matches, "relayer_url").unwrap_or_else(|| {
-            value_of(&matches, "relayer_auth_service_address").expect("missing relayer-url")
-        });
-        let mut auth_service_endpoint =
-            Endpoint::from_shared(addr.clone()).expect("invalid relayer-url value");
-        if addr.contains("https") {
-            auth_service_endpoint = auth_service_endpoint
-                .tls_config(tonic::transport::ClientTlsConfig::new())
-                .expect("failed to set tls_config");
+    let mut block_engine_config = BlockEngineConfig::default();
+    if matches.is_present("block_engine_url") {
+        block_engine_config.block_engine_url =
+            value_of(&matches, "block_engine_url").expect("couldn't parse block_engine_url");
+    } else {
+        let error_msg = "Specifying seperate auth and backend addresses for block engine is deprecated.  Recommended to use --block_engine_url instead.\
+                    If using block_engine_auth_service_address and block_engine_address, they must both be provided and set to the same value.";
+        match (
+            matches.is_present("block_engine_auth_service_address"),
+            matches.is_present("block_engine_address"),
+        ) {
+            (true, true) => {
+                let auth_addr: String = value_of(&matches, "block_engine_auth_service_address")
+                    .expect("couldn't parse block_engine_auth_service_address");
+                let backend_addr: String = value_of(&matches, "block_engine_address")
+                    .expect("couldn't parse block_engine_address");
+                if auth_addr != backend_addr {
+                    eprintln!("{}", error_msg);
+                    exit(1);
+                } else {
+                    block_engine_config.block_engine_url = backend_addr;
+                }
+            }
+            (false, false) => {}
+            _ => {
+                eprintln!("{}", error_msg);
+                exit(1);
+            }
         }
+    }
+    block_engine_config.trust_packets = matches.is_present("trust_block_engine_packets");
```

Comment:
nit: move to beginning of the if statement after block_engine_config instantiation.